### PR TITLE
[docs] Add diagrams to docs

### PR DESCRIPTION
**Pull request by kinow**
## Description

Adding diagrams created when analyzing a CWL Viewer issue: https://github.com/common-workflow-language/cwlviewer/issues/424#issuecomment-1160893596

## Motivation and Context

For myself and for future developers.

## How Has This Been Tested?

NA

## Screenshots (if appropriate):

See linked issue or preview by opening the PNG files.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by mr-c** [comment]
Thanks!

How did you create these diagrams? Can we add the sources?

File: docs/diagrams/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Diagrams
+
+This folder contains diagrams about functionalities of CWL Viewer.
+The source file for the diagrams is stored in a SVG file. We suggest
+contributors to use Inkscape to edit the SVG file.
```
Comment (kinow) [inline docs/diagrams/README.md:5]: @mr-c  :point_up:

Comment (kinow) [reply docs/diagrams/README.md:5]: The source is included as `docs/diagrams/cwlviewer.svg`

Comment (mr-c) [reply docs/diagrams/README.md:5]: Did you make this by hand using inkscape? Or is this SVG an export from some other tool?

Comment (kinow) [reply docs/diagrams/README.md:5]: By hand.

**Review by mr-c — APPROVED** [review]

**Pull request merged**